sbc_translation/models/correspondence.py

```diff
@@ -68,12 +69,33 @@ def process_letter(self):
          needed and upload to translation platform. """
         for letter in self:
             if letter.translation_language_id not in \
-                    letter.supporter_languages_ids:
+                    letter.supporter_languages_ids or not \
+                    self.is_writed_in_language(letter.translated_text,
+                                               letter.translation_language_id):
                 letter.download_attach_letter_image()
                 letter.send_local_translate()
             else:
                 super(Correspondence, letter).process_letter()
 
+    def is_writed_in_language(self, text, language_id):
+        """ Detect if text is writed in the language corresponding to the
+        language_id """
+        if text is not None and language_id is not None:
+            s = text.strip(' \t\n\r')
+            if s != "":
+                # find the language name of text argument
+                detectlanguage.configuration.api_key = \
+                    "ec1377a3d666c2ff8c60a252e2b9ec2f"
```

Comment:
Just I seen then when I pushed the code and I thought too that it would be good to change...